W3C.Domain/IdentificationService/IdentificationServiceClient.cs

```diff
@@ -0,0 +1,70 @@
+using System;
+using System.Collections.Generic;
+using System.IdentityModel.Tokens.Jwt;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.IdentityModel.Tokens;
+using Newtonsoft.Json;
+using W3C.Contracts.Admin.Permission;
+
+namespace W3C.Domain.IdentificationService;
+
+public class IdentificationServiceClient
+{
+    private readonly HttpClient _httpClient;
+    private readonly string _identificationSecret = Environment.GetEnvironmentVariable("WEBSITE_BACKEND_TO_ID_SERVICE_SECRET");
+    private static readonly string ServiceApiUrl = Environment.GetEnvironmentVariable("IDENTIFICATION_API") ?? "http://localhost:8081";
+    private string _cachedToken;
+    private DateTime _tokenExpiryTime;
+
+    public IdentificationServiceClient(IHttpClientFactory httpClientFactory)
```

Comment:
Can we write some tests for the functions in this service client?

Reply:
added